app/Event/ProductionControllerProvider.php

```diff
@@ -34,6 +36,15 @@ function (Application $app) {
             }
         );
 
+        $app[ProductionSuggestionController::class] = $app->share(
+            function (Application $app) {
+                return new ProductionSuggestionController(
+                    $app[SimilaritiesClient::class],
+                    $app[ProductionEnrichedEventRepository::class]
```

Comment:
I'd suggest to always use `event_jsonld_repository`. The various decorators around it are then more of an implementation detail. The extra `BroadcastingDocumentRepositoryDecorator` around it should be no problem, that only dispatches an event on the event bus when the JSON-LD gets _saved_. When fetching the JSON-LD it does nothing that could cause issues, and the performance overhead should be negligable.

Reply:
Additionally, you can then move the `ProductionEnrichedEventRepository::class` definition back inside the `event_jsonld_repository`. That makes it a bit easier to read, especially with multiple decorators.

Reply:
done